test/ZendeskApi_v2.Test/AttachmentTests.cs

```diff
@@ -65,5 +66,47 @@ public async Task CanDowloadAttachment()
             Assert.That(api.Tickets.Delete(t1.Ticket.Id.Value), Is.True);
             Assert.That(api.Attachments.DeleteUpload(res));
         }
+
+        [Test]
+        public async Task CanRedactAttachment()
+        {
+            //This could probably be brought into a helper for above two tests perhaps
+            var path = Path.Combine(TestContext.CurrentContext.TestDirectory, "testupload.txt");
+
+            var res = await api.Attachments.UploadAttachmentAsync(new ZenFile()
+            {
+                ContentType = "text/plain",
+                FileName = "testupload.txt",
+                FileData = File.ReadAllBytes(path)
+            });
+
+            var ticket = new Ticket()
+            {
+                Subject = "testing attachments",
+                Priority = TicketPriorities.Normal,
+                Comment = new Comment()
+                {
+                    Body = "comments are required for attachments",
+                    Public = true,
+                    Uploads = new List<string>() { res.Token }
+                },
+            };
+
+            var t1 = await api.Tickets.CreateTicketAsync(ticket);
+
+            var comments = api.Tickets.GetTicketComments(t1.Ticket.Id.Value);
+
+            var attach = comments.Comments[0].Attachments[0];
+
+            var delRes = api.Attachments.RedactCommentAttachment(attach.Id, t1.Ticket.Id.Value, comments.Comments[0].Id.Value);
+            //Returned correct attachment
+            Assert.That(delRes.Attachment.Id, Is.EqualTo(attach.Id));
+
+            var ex = Assert.Throws<WebException>(() => api.Attachments.RedactCommentAttachment(attach.Id, t1.Ticket.Id.Value, comments.Comments[0].Id.Value));
```

Comment:
How about we get the tickets comments again and check the text is correct that way we are not throwing an exception.

Reply:
Yeah no probs will check the attachment is redacted/replaced as per Zendesk's docs.